Add Half-Life 2 Deathmatch

Bindon10 opened the pull request:
Introduce Half-Life 2 Deathmatch Dedicated Server template (Not Half-Life Deathmatch Source)

On Linux it is required to be run in a docker container (Wine 8) or have GLIBC_2.29 installed on the host

Added the ability to change Metamod/Sourcemod branches independently

Greelan commented:
Thanks. Can you make the metamod/sourcmod settings consistent with the L4D2 template that has just been merged. Consistency helps with maintenance. (I've separately updated other templates that have metamod/sourcemod settings but in a slightly different way for backwards compatibility given they were existing templates.)

I suggest the category name not include dashes - looks odd. 

Have all the settings in server.cfg been tested to verify they work? I have found that srcds/hlds servers are not very consistent

Bindon10 commented:
> Thanks. Can you make the metamod/sourcmod settings consistent with the L4D2 template that has just been merged. Consistency helps with maintenance. (I've separately updated other templates that have metamod/sourcemod settings but in a slightly different way for backwards compatibility given they were existing templates.)
> 
> I suggest the category name not include dashes - looks odd.
> 
> Have all the settings in server.cfg been tested to verify they work? I have found that srcds/hlds servers are not very consistent

As far as i'm aware, i've made the settings consistent with the L4D2 Template, though I did have to add the 2.0 version for Metamod as that's the build that works for the 20th anniversary update (for HL2/HL2DM) in my latest push.

Fixed the dashes, completely forgot to change it before the pull request, it now shows as "Half life 2 Deathmatch"

The settings as far as my testing has been verified to work, the only setting i'm aware of that does not work correctly is mp_falldamage, though that is fixed by a sourcemod plugin. The option is there purely in the event that they install a certain SM plugin (https://github.com/utharper/sourcemod-hl2dm#xFix)

Greelan commented:
Nice, I will check it out.

Greelan commented:
A few preliminary comments:

- I'd really encourage you to organise the settings and server.cfg in a consistent way with the other srcds/hlds templates. Helps the user to have the same settings in the same places. I acknowledge that this server has some additional settings
- It helps for the server.cfg to be in the same order
- Some of the settings in the Server subcategory seem more appropriate for Gameplay
- I don't understand the point of having commented settings in server.cfg
- Similarly, it's not practical to require some settings to be amended manually via AMP_server.cfg. Why can't they be in the UI?
- No SourceTV/HLTV for this server?

Bindon10 commented:
I tried my best to keep this revision on par with how the other srcds games are organized but i'm sure there's some things that are out of place that I didn't know exactly what to do with or missed entirely. 

- Added SourceTV Settings, as far as I am aware; all of these settings work as intended. Limited testing by myself is hard.
- Reorganized and cleaned various settings that were either not needed or just didn't do anything

Greelan commented:
I'm about 85% through my review....
